tket/tests/Circuit/test_Boxes.cpp

```diff
@@ -149,6 +157,38 @@ SCENARIO("Using Boxes", "[boxes]") {
     // check it's the identity
     REQUIRE((d1m - Eigen::Matrix4cd::Identity()).cwiseAbs().sum() < ERR_EPS);
   }
+  GIVEN("Unitary Box Identity constructors") {
```

Comment:
I don't see what the circuit equality tests here are testing. The API doesn't promise that an empty UnitaryBox will decompose to these specific circuits, and a change of implementation could change that and so make this test fail. I think it's only valid to check unitary equivalence of the decomposed circuits.

(I also don't see what it has to do with the changes in this PR?)

Reply:
This was my attempt to please the CI

Reply:
OK I see, but even so I don't think it makes sense to `REQUIRE` these exact decompositions, since we don't promise any specific circuit here. (Test coverage would not be affected by removing the circuit-equality checks.)